### PR TITLE
[Feature] Order Chat

### LINKED ISSUE #90: Chat de pedidos

### PR DESCRIPTION
**Pull request by mtullyoc**
## Summary

- Design fixed
- Auto fetch older messages onEndReached

Resolves #90 

https://user-images.githubusercontent.com/17722915/115990360-2c7e8300-a599-11eb-8903-5d3513f82af4.MP4

### CODE REVIEW

File: src/screens/OrderChat/components/MessageItem.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import * as S from '../styles';
+
+export const TIME_FORMAT = 'HH:mm - dd/LL/yyyy';
+
+interface MessageItemProps {
+  isImSender: boolean;
+  message: any;
+}
+
+export const MessageItem: React.FC<MessageItemProps> = ({ isImSender, message }) => (
```
Comment (mateus4k) [inline src/screens/OrderChat/components/MessageItem.tsx]: The `FC` type has some side effects, like always accepting a child as a prop.
```suggestion
export const MessageItem = ({ isImSender, message }: MessageItemProps) => (
```
Read more [here](https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/)

File: src/screens/OrderChat/index.tsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+      if (!chatInstance) {
+        return;
+      }
+
+      let m;
+
+      await getChatMessages(chatInstance.id, { offset: 0 }).then((data) => {
+        m = merge(messages, data, 'id');
+        m = m.sort((a: any, b: any) => b.id - a.id);
```
Comment (mateus4k) [inline src/screens/OrderChat/index.tsx]: This lines are not looking so good. What does the merge function do? I think a reduce function would solve a lot of things here 🤔

File: src/hooks/useKeyboard.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { Keyboard } from 'react-native';
+
+export const useKeyboard = (): [boolean, () => void] => {
+  const [visible, setVisible] = React.useState(false);
```
Comment (mateus4k) [inline src/hooks/useKeyboard.tsx:5]: Nice implementation. Congrats! 😎

File: src/screens/OrderChat/index.tsx (hunk 2 above)
Comment (mtullyoc) [reply src/screens/OrderChat/index.tsx]: I did some refactoring with more descriptive names and simplified logic. Thanks!

**Pull request merged**